controllers/provider-azure/pkg/webhook/controlplane/ensurer.go

```diff
@@ -180,15 +185,19 @@ func (e *ensurer) ensureChecksumAnnotations(ctx context.Context, template *corev
 func (e *ensurer) EnsureKubeletServiceUnitOptions(ctx context.Context, ectx genericmutator.EnsurerContext, opts []*unit.UnitOption) ([]*unit.UnitOption, error) {
 	if opt := extensionswebhook.UnitOptionWithSectionAndName(opts, "Service", "ExecStart"); opt != nil {
 		command := extensionswebhook.DeserializeCommandLine(opt.Value)
-		command = ensureKubeletCommandLineArgs(command)
+		command = e.ensureKubeletCommandLineArgs(ctx, ectx, command)
 		opt.Value = extensionswebhook.SerializeCommandLine(command, 1, " \\\n    ")
 	}
 	return opts, nil
 }
 
-func ensureKubeletCommandLineArgs(command []string) []string {
+func (e *ensurer) ensureKubeletCommandLineArgs(ctx context.Context, ectx genericmutator.EnsurerContext, command []string) []string {
 	command = extensionswebhook.EnsureStringWithPrefix(command, "--cloud-provider=", "azure")
 	command = extensionswebhook.EnsureStringWithPrefix(command, "--cloud-config=", "/var/lib/kubelet/cloudprovider.conf")
+
+	if _, exists, _ := e.getAcrConfigMap(ctx, ectx); exists {
```

Comment:
Why is the error not checked? Now, it would silently not inject the flag if an error occurs.